cdn_lambda/functions/map_to_s3.py

```diff
@@ -0,0 +1,61 @@
+import json
+import logging
+import os
+from datetime import datetime
+
+import boto3
+
+LOG = logging.getLogger("map-to-s3-lambda")
+
+
+def lambda_handler(event, context):
+    _ = context
+
+    # This AWS Lambda function must be deployed with DB_TABLE_NAME and
+    # DB_TABLE_REGION environment variables
+    table_name = os.getenv("DB_TABLE_NAME")
+    table_region = os.getenv("DB_TABLE_REGION")
+
+    # Get uri from origin request event
+    request = event["Records"][0]["cf"]["request"]
+    web_uri = request["uri"]
+
+    # Query latest item with matching uri from DynamoDB table
+    LOG.info("Querying '%s' table for '%s'...", table_name, web_uri)
+
+    query_result = boto3.client("dynamodb", region_name=table_region).query(
+        TableName=table_name,
+        Limit=1,
+        ScanIndexForward=False,
+        KeyConditionExpression="web_uri = :u and from_date <= :d",
+        ExpressionAttributeValues={
+            ":u": {"S": web_uri},
+            ":d": {"S": str(datetime.utcnow())},
```

Comment:
The default str formatting of datetime here doesn't seem ideal, it doesn't seem to be exactly one of the standard ISO8601 formats (e.g. the same format Pulp uses).  I played around with it a bit and I think the best way to get our timestamps is like this:

```
datetime.now(timezone.utc).isoformat(timespec='milliseconds')
```

- gives a value like 2020-02-13T04:20:40.444+00:00 which follows ISO8601 standard, and includes timezone so there's no ambiguity
- datetime.now(timezone.utc) is recommended over datetime.utcnow (according to https://docs.python.org/3.8/library/datetime.html)
- The default behavior of str/isoformat is "timespec='auto': Same as 'seconds' if microsecond is 0, same as 'microseconds' otherwise."  This seems to be undesirable, as datetimes which happen to land on a microsecond=0 (very small chance) will happen to have a different length from all other datetimes.  So there will be a small number of timestamps which use a different format for no real reason.